File: 2404-most-frequent-even-element/2404-most-frequent-even-element.py
class Solution:
    def mostFrequentEven(self, nums: List[int]) -> int:
        freq = {}
        mostFrequent = -1
        max_frequency = -1
        for num in nums:
            if num % 2 == 0:
                freq[num] = freq.get(num, 0)+1
            
        for key,value in freq.items():
            if value > max_frequency or value == max_frequency and key < mostFrequent:
                mostFrequent = key
                max_frequency = value
        return mostFrequent

File: 2404-most-frequent-even-element/2404-most-frequent-even-element.java
class Solution {
    public int mostFrequentEven(int[] nums) {
        int m=Integer.MIN_VALUE;
        for(int num:nums){
            m=Math.max(m,num);
        }
        int ans=-1;
        int[] map=new int[m+1];
        for(int num:nums){
            if(num%2==0){
                map[num]++;
            }
        }
        int max=Integer.MIN_VALUE;
        for(int i=m;i>=0;i--){
            if(map[i]>=max && map[i]!=0){
                max=map[i];
                ans=i;
            }
        }
        return ans;
    }
}